File: 02.実践帳票サンプル集－様々な帳票手法/VB/Common/Make郵便番号一覧.vb
Imports Pao.Reports
Imports System
Imports System.Data
Imports System.Data.OleDb
Imports System.Windows

Namespace Sample
    Friend Module Make郵便番号一覧
        Public Sub SetupData(paoRep As IReport)
            'Excel データベースへの接続
            Dim connection As OleDbConnection = Util.ConnectExcelDB("zip.xls")
            'データセットへテーブルをセットする。ヘッダと明細の2テーブル
            Dim SQL As String = "select * from [郵便番号テーブル$]"
            Dim dataAdapter As New OleDbDataAdapter(SQL, connection)
            Dim ds As New DataSet()

            dataAdapter.Fill(ds, "PostTable")

            Dim table As DataTable = ds.Tables("PostTable")
            Dim page As Integer = 0
            Dim line As Integer = 999
            Dim hDate As String = System.DateTime.Now.ToString()

            paoRep.LoadDefFile(Util.SharePath & "zip1.prepd")

            For Each row As DataRow In table.Rows
                line += 1
                If line > 32 Then
                    ' Head Print
                    If page <> 0 Then paoRep.PageEnd()
                    page += 1

                    If page = 6 Then
                        paoRep.LoadDefFile(Util.SharePath & "zip2.prepd")
                    End If

                    paoRep.PageStart()
                    paoRep.Write("日時", hDate)
                    paoRep.Write("ページ", "Page-" & page.ToString())

                    'QRコード描画
                    If page < 6 Then
                        paoRep.Write("QR", row("郵便番号").ToString() & " " & row("市区町村").ToString() & row("住所").ToString())
                    End If

                    line = 1
                End If

                'Body Print
                paoRep.Write("郵便番号", row("郵便番号").ToString(), line)
                paoRep.Write("市区町村", row("市区町村").ToString(), line)
                paoRep.Write("住所", row("住所").ToString(), line)
                paoRep.Write("横罫線", line)

                If page > 5 AndAlso line Mod 2 = 0 Then
                    paoRep.Write("網掛け", line \ 2)
                End If
            Next

            paoRep.PageEnd()
            Return
        End Sub
    End Module
End Namespace

File: 02.実践帳票サンプル集－様々な帳票手法/CS/Common/Make郵便番号一覧.cs
using Pao.Reports;
using System;
using System.Data;
using System.Data.OleDb;

namespace Sample
{
    static class Make郵便番号一覧
    {
        public static void SetupData(IReport paoRep)
        {
            //Excel データベースへの接続
            OleDbConnection connection = Util.ConnectExcelDB("zip.xls");

            //データセットへテーブルをセットする。ヘッダと明細の2テーブル
            string SQL = "select * from [郵便番号テーブル$]";

            OleDbDataAdapter dataAdapter = new OleDbDataAdapter(SQL, connection);
            DataSet ds = new DataSet();
            dataAdapter.Fill(ds, "PostTable");
            DataTable table = ds.Tables["PostTable"];

            int page = 0;
            int line = 999;
            string hDate = System.DateTime.Now.ToString();

            paoRep.LoadDefFile(Util.SharePath + "zip1.prepd");
            foreach (DataRow row in table.Rows)
            {
                line++;
                if (line > 32)
                { // Head Print
                    if (page != 0) paoRep.PageEnd();

                    page++;

                    if (page == 6)
                    {
                        paoRep.LoadDefFile(Util.SharePath + "zip2.prepd");
                    }

                    paoRep.PageStart();

                    paoRep.Write("日時", hDate);
                    paoRep.Write("ページ", "Page-" + page.ToString());

                    //QRコード描画
                    if (page < 6)
                    {
                        paoRep.Write("QR", row["郵便番号"].ToString() + " " + row["市区町村"].ToString() + row["住所"].ToString());
                    }

                    line = 1;

                }

                //Body Print
                paoRep.Write("郵便番号", row["郵便番号"].ToString(), line);
                paoRep.Write("市区町村", row["市区町村"].ToString(), line);
                paoRep.Write("住所", row["住所"].ToString(), line);
                paoRep.Write("横罫線", line);


                if (page > 5 && line % 2 == 0)
                    paoRep.Write("網掛け", line / 2);

            }
            paoRep.PageEnd();

            return;
        }
    }
}
